dev/CommandBarFlyout/CommandBarFlyoutCommandBar.cpp

```diff
@@ -85,75 +141,187 @@ void CommandBarFlyoutCommandBar::OnApplyTemplate()
     m_openingStoryboard.set(GetTemplateChildT<winrt::Storyboard>(L"OpeningStoryboard", thisAsControlProtected));
     m_closingStoryboard.set(GetTemplateChildT<winrt::Storyboard>(L"ClosingStoryboard", thisAsControlProtected));
 
+    if (auto moreButton = m_moreButton.get())
+    {
+        // Initially only the first focusable primary and secondary commands
+        // keep their IsTabStop set to True.
+        if (moreButton.IsTabStop())
+        {
+            moreButton.IsTabStop(false);
+        }
+    }
+
     AttachEventHandlers();
     UpdateFlowsFromAndFlowsTo();
     UpdateUI(false /* useTransitions */);
 }
 
-void CommandBarFlyoutCommandBar::SetOwningFlyout(winrt::CommandBarFlyout const& owningFlyout)
-{
+void CommandBarFlyoutCommandBar::SetOwningFlyout(
+    winrt::CommandBarFlyout const& owningFlyout)
+{   
     m_owningFlyout = owningFlyout;
 }
 
 void CommandBarFlyoutCommandBar::AttachEventHandlers()
 {
-    // In addition to closing the CommandBar if someone hits the escape key,
-    // we also want to close the whole flyout, as well.
-    m_keyDownHandler = winrt::box_value<winrt::KeyEventHandler>({
-        [this](auto const&, winrt::KeyRoutedEventArgs const& args)
+    COMMANDBARFLYOUT_TRACE_INFO(*this, TRACE_MSG_METH, METH_NAME, this);
+
+    if (!SharedHelpers::IsRS3OrHigher())
+    {
+        // Prior to RS3, UIElement.PreviewKeyDown is not available. Thus IsTabStop
+        // for SecondaryCommands is left to True and the KeyDown event is used to
+        // close the whole flyout with the escape key.
+        m_keyDownHandler = winrt::box_value<winrt::KeyEventHandler>(
         {
```

Comment:
I just added a helper in dev\inc\RoutedEventHelpers.h that I'd recommend instead to help make this a "revoker" pattern as well as has a pattern to make the pairing between the event and the handler be type-safe.

Reply:
Done. Now using RoutedEventHandler_revoker. Thanks.